### PR TITLE
reactive removal from common/rest and vault

### LINKED ISSUE #6849: Update Common REST API to blocking only

### PR DESCRIPTION
**Pull request by arjav-desai**
Fixes #6849 , #6911 , #7029 and #7030

### CODE REVIEW

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java [hunk 1]
```diff
@@ -376,7 +360,9 @@ protected B me() {
          * This implementation builds the web client and sets up JSON factories.
          */
         protected void preBuild() {
-            webClientBuilder.addMediaSupport(JsonpSupport.create());
+            webClientBuilder.mediaContext(MediaContext.builder()
+                                                      .addMediaSupport(JsonpSupport.create())
```
Comment (romain-grecourt) [inline integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java:364]: The `addMediaSupport` method was added recently, you can now do this:

```java
Http1Client.builder()
           .addMediaSupport(JsonpSupport.create())
           .build();
```

I.e. you can leave this method unchanged

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiOptionalResponse.java
Comment (romain-grecourt) [inline integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiOptionalResponse.java:1]: This file contains only formatting changes.

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java [hunk 2]
```diff
@@ -47,42 +46,51 @@ class AppRoleRestApi extends VaultRestApi {
...
-                    requestBuilder.headers().add("X-Vault-Token", token.token());
-                    return requestBuilder;
-                })
-                .onError(throwable -> LOGGER.log(Level.WARNING, "Failed to renew AppRole token", throwable));
+        try {
+            // we need to renew the token - this may be a concurrent operation as we do not care who wins
+            Login.Response response = auth.login(Login.Request.create(appRoleId, secretId));
+            VaultTokenBase token = response.token();
+            this.currentToken.set(token);
+            requestBuilder.headers(headers -> {
```
Comment (romain-grecourt) [inline integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java:73]: We should make sure that we add a shortcut for this scenario.
Adding a single header with string/string should not be that complicated.

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java [hunk 3]
```diff
@@ -85,55 +88,57 @@ public Optional<RestApi> authenticate(Config config, Vault.Builder vaultBuilder)
...
-                                   })
-                                   .faultTolerance(vaultBuilder.ftHandler())
-                                   .auth(loginVault.auth(AppRoleAuthRx.AUTH_METHOD, methodPath))
-                                   .appRoleId(appRoleId)
-                                   .secretId(secretId)
-                                   .build());
+                                         .webClientBuilder(webclient -> {
+                                             webclient.baseUri(address + "/v1");
+                                             vaultBuilder.baseNamespace()
+                                                         .ifPresent(ns -> webclient.header(Http.Header.create("X-Vault-Namespace"), ns));
```
Comment (romain-grecourt) [inline integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java:129]: Same comment as above, we should overload the `header` method to accept a string for the header name.

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java [hunk 4]
```diff
@@ -46,39 +45,43 @@ static Builder k8sBuilder() {
...
+                                                            Http.Method method,
+                                                            String requestId) {
         VaultTokenBase k8sToken = currentToken.get();
 
         if (k8sToken != null) {
             if (!k8sToken.renewable() || k8sToken.created().plus(k8sToken.leaseDuration()).isAfter(Instant.now())) {
-                requestBuilder.headers().add("X-Vault-Token", k8sToken.token());
-                return Single.just(requestBuilder);
+                requestBuilder.headers(headers -> {
+                    headers.add(Http.Header.create("X-Vault-Token"), k8sToken.token());
```
Comment (romain-grecourt) [inline integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java:58]: header shortcut

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java [hunk 5]
```diff
@@ -46,39 +45,43 @@ static Builder k8sBuilder() {
...
-                .map(it -> {
-                    VaultTokenBase token = it.token();
-                    currentToken.set(token);
-                    requestBuilder.headers().add("X-Vault-Token", token.token());
-                    return requestBuilder;
-                });
+        Login.Response response = auth.login(Login.Request.create(roleName, jwtToken));
+        VaultTokenBase token = response.token();
+        currentToken.set(token);
+        requestBuilder.headers(headers -> {
```
Comment (romain-grecourt) [inline integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java:69]: header shortcut

File: integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java [hunk 6]
```diff
@@ -77,26 +78,27 @@ public Optional<RestApi> authenticate(Config config, Vault.Builder vaultBuilder)
...
-                            .or(vaultBuilder::baseNamespace)
-                            .ifPresent(ns -> builder.addHeader("X-Vault-Namespace", ns));
-                    vaultBuilder.webClientUpdater().accept(builder);
-                })
-                .faultTolerance(vaultBuilder.ftHandler())
-                .build();
+                           .webClientBuilder(builder -> {
+                               builder.config(vaultBuilder.config().get("webclient"))
+                                      .baseUri(address + "/v1")
+                                      .header(Http.Header.create("X-Vault-Token"), token);
```
Comment (romain-grecourt) [inline integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java:90]: overloaded header method

File: integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java [hunk 7]
```diff
@@ -77,26 +78,27 @@ public Optional<RestApi> authenticate(Config config, Vault.Builder vaultBuilder)
...
+                               Optional.ofNullable(baseNamespace)
+                                       .or(vaultBuilder::baseNamespace)
+                                       .ifPresent(ns -> builder.header(Http.Header.create("X-Vault-Namespace"), ns));
```
Comment (romain-grecourt) [inline integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java:93]: overloaded header method

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java [hunk 8]
```diff
@@ -98,6 +99,32 @@ default B header(Http.HeaderName name, String value) {
      */
     B headers(Function<ClientRequestHeaders, WritableHeaders<?>> headersConsumer);
 
+    /**
```
Comment (romain-grecourt) [inline nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java:102]: These two methods should be added in a separate PR.

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/WebClient.java
Comment (romain-grecourt) [inline nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/WebClient.java:1]: All changes to the WebClient should be done in separate PR(s).
See also:
- https://github.com/helidon-io/helidon/issues/6935
- https://github.com/helidon-io/helidon/issues/7034

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1Client.java
Comment (romain-grecourt) [inline nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1Client.java:1]: Same as above, WebClient changes should be done separately.

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java (hunk 2 above)
Comment (tomas-langer) [reply integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java:73]: Actually this is exactly the case where we SHOULD NEVER use `String` here. Please create a constant `HeaderName` for `X-Vault-Token` and use it everywhere instead of using a shortcut

File: integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java (hunk 6 above)
Comment (tomas-langer) [reply integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java:90]: use constant for `HeaderName`, do not create it from a constant string, do not replace it with `String` shortcut

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiJsonRequest.java [hunk 9]
```diff
@@ -20,12 +20,12 @@
...
 import java.util.Optional;
 
-import io.helidon.common.media.type.MediaType;
+import io.helidon.common.http.HttpMediaType;
 
 /**
  * Common base class for REST requests that have an entity.
  * This class acts as a mutable builder of request JSON object.
- *
+ * <br>
```
Comment (tomas-langer) [inline integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiJsonRequest.java:28]: please use `<p>`, not <br>

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiRestRequest.java [hunk 10]
```diff
@@ -22,22 +22,22 @@
...
 
-import io.helidon.common.media.type.MediaType;
+import io.helidon.common.http.HttpMediaType;
 
 /**
  * Common base class for REST requests.
  * This class acts as a mutable builder without a build method, as the intended use is to pass it
  * to a {@code io.helidon.integrations.common.rest.RestApi}, not to send it around for parallel processing.
- *
+ * <br>
```
Comment (tomas-langer) [inline integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiRestRequest.java:31]: Use <p> instead

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java [hunk 11]
```diff
@@ -276,16 +262,15 @@ Single<T> invokeOptional(Method method,
      * @param <T> type of the built {@link io.helidon.integrations.common.rest.RestApi}
      */
     abstract class Builder<B extends Builder<B, T>, T extends RestApi> implements io.helidon.common.Builder<B, T> {
-        private final WebClient.Builder webClientBuilder = WebClient.builder()
-                .followRedirects(true)
-                .keepAlive(true);
+        private final Http1ClientBuilder webClientBuilder = Http1Client.builder();
+        //.followRedirects(true)
```
Comment (tomas-langer) [inline integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java:266]: Is there a follow up issue to fix this? Not mentioned in the PR

File: integrations/pom.xml [hunk 12]
```diff
@@ -62,6 +62,8 @@
                     <configuration>
                         <compilerArgs>
                             <arg>-Xlint:all</arg>
+                            <arg>-Xlint:-missing-explicit-ctor</arg>
```
Comment (tomas-langer) [inline integrations/pom.xml:65]: Missing explicit constructor is a good warning, we should always have explicit constructors (there are two reasons - javadoc if it should be there, pointing out an error if it should not be there, such as when you forget to add a private constructor for a `Builder`)

File: integrations/pom.xml [hunk 13]
```diff
@@ -62,6 +62,8 @@
...
+                            <arg>-Xlint:-serial</arg>
```
Comment (tomas-langer) [inline integrations/pom.xml:66]: Why is this warning disabled?

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java (hunk 2 above)
Comment (tomas-langer) [reply integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java:73]: And of course if we are missing a `header(HeaderName, String)` and `header(HeaderValue)` on request builder, it would be good to add them

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java [hunk 14]
```diff
@@ -68,15 +69,17 @@ public static Builder builder() {
...
+                                .orElse(true);
 
         if (!enabled) {
             return Optional.empty();
         }
         Optional<String> maybeAppRoleId = Optional.ofNullable(this.appRoleId)
-                .or(() -> config.get("auth.app-role.role-id")
-                        .asString()
-                        .asOptional());
+                                                  .or(() -> config.get("auth.app-role.role-id")
```
Comment (tomas-langer) [inline integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java:80]: I think you have a different configuration of formatting than the one in `etc` in the project. Could you validate it? I am not a big fan of reformatting statements such as this one, as on the next PR by somebody else, it may be reformatted back

Comment (tomas-langer) [reply integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java:80]: this is a general comment to everything in this PR

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java (hunk 3 above)
Comment (tomas-langer) [reply integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java:129]: Again, please create a constant `HeaderName` for namespace and use it.

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java [hunk 15]
```diff
@@ -98,6 +99,32 @@ default B header(Http.HeaderName name, String value) {
...
+        });
+    }
+
+    /**
+     * Sets the MIME type of the response body.
+     *
+     * @param contentType Media type of the content.
+     * @return updated request
+     */
+    default B contentType(HttpMediaType contentType) {
```
Comment (tomas-langer) [inline nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java:121]: The javadoc is wrong - this sets the content type of the request. Accept header is the one to set expected response types.

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1Client.java [hunk 16]
```diff
@@ -96,6 +99,12 @@ public Http1ClientBuilder baseUri(URI baseUri) {
             return this;
         }
 
+//        public Http1ClientBuilder config(Config config) {
```
Comment (tomas-langer) [inline nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1Client.java:102]: Please remove commented out code.

**Review by tomas-langer — DISMISSED** [review]

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java (hunk 11 above)
Comment (romain-grecourt) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java:266]: https://github.com/helidon-io/helidon/issues/6804 which is already fixed

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java (hunk 1 above)
Comment (arjav-desai) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java]: Updated it accordingly.

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java (hunk 2 above)
Comment (arjav-desai) [reply integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleRestApi.java]: I had added new method `header(HeaderName, String)` ; and using it here now.

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java (hunk 3 above)
Comment (arjav-desai) [reply integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java]: Updated.

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java (hunk 4 above)
Comment (arjav-desai) [reply integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java]: Updated.

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java (hunk 5 above)
Comment (arjav-desai) [reply integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java:71]: Updated.

File: integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java (hunk 7 above)
Comment (arjav-desai) [reply integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java]: Updated.

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java (hunk 8 above)
Comment (arjav-desai) [reply nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java:102]: Okay, will move webclient files to separate PR

File: integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java (hunk 6 above)
Comment (arjav-desai) [reply integrations/vault/auths/token/src/main/java/io/helidon/integrations/vault/auths/token/TokenVaultAuth.java]: Updated.

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/WebClient.java
Comment (arjav-desai) [reply nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/WebClient.java:1]: I have taken #7034 and will close that with separate webclient PR.

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiJsonRequest.java (hunk 9 above)
Comment (arjav-desai) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiJsonRequest.java]: Updated.

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiOptionalResponse.java
Comment (arjav-desai) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiOptionalResponse.java:1]: Updated.

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiRestRequest.java (hunk 10 above)
Comment (arjav-desai) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiRestRequest.java]: Updated to use `<p>`

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java (hunk 11 above)
Comment (arjav-desai) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java]: Added redirect and keepalive.

File: integrations/pom.xml (hunk 12 above)
Comment (arjav-desai) [reply integrations/pom.xml]: Removed it.

File: integrations/pom.xml (hunk 13 above)
Comment (arjav-desai) [reply integrations/pom.xml]: Removed it.

File: integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java (hunk 14 above)
Comment (arjav-desai) [reply integrations/vault/auths/approle/src/main/java/io/helidon/integrations/vault/auths/approle/AppRoleVaultAuth.java]: Have used idea formatter provided in slack and used that across PR.

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java (hunk 15 above)
Comment (arjav-desai) [reply nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/ClientRequest.java:121]: Updated javadoc

File: nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1Client.java (hunk 16 above)
Comment (arjav-desai) [reply nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1Client.java]: Deleted it.

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java [hunk 17]
```diff
@@ -276,16 +262,13 @@ Single<T> invokeOptional(Method method,
      * @param <T> type of the built {@link io.helidon.integrations.common.rest.RestApi}
      */
     abstract class Builder<B extends Builder<B, T>, T extends RestApi> implements io.helidon.common.Builder<B, T> {
-        private final WebClient.Builder webClientBuilder = WebClient.builder()
-                .followRedirects(true)
-                .keepAlive(true);
+        private final Http1ClientBuilder webClientBuilder = Http1Client.builder().followRedirect(true).keepAlive(true);
```
Comment (romain-grecourt) [inline integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java:265]: Break on multiple lines, so that we only see the first line changed.

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java [hunk 18]
```diff
@@ -46,39 +47,43 @@ static Builder k8sBuilder() {
...
+                                                            ApiRequest<?> request,
+                                                            Http.Method method,
+                                                            String requestId) {
         VaultTokenBase k8sToken = currentToken.get();
 
         if (k8sToken != null) {
             if (!k8sToken.renewable() || k8sToken.created().plus(k8sToken.leaseDuration()).isAfter(Instant.now())) {
-                requestBuilder.headers().add("X-Vault-Token", k8sToken.token());
-                return Single.just(requestBuilder);
+                requestBuilder.headers(headers -> {
```
Comment (romain-grecourt) [inline integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java:59]:
```java
requestBuilder.header(VAULT_TOKEN_HEADER_NAME, k8sToken.token());
```

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java [hunk 19]
```diff
@@ -46,39 +47,43 @@ static Builder k8sBuilder() {
...
-                .map(it -> {
-                    VaultTokenBase token = it.token();
-                    currentToken.set(token);
-                    requestBuilder.headers().add("X-Vault-Token", token.token());
-                    return requestBuilder;
-                });
+        Login.Response response = auth.login(Login.Request.create(roleName, jwtToken));
+        VaultTokenBase token = response.token();
+        currentToken.set(token);
+        requestBuilder.headers(headers -> {
```
Comment (romain-grecourt) [inline integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java:71]: requestBuilder.header(VAULT_TOKEN_HEADER_NAME, token.token());

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sVaultAuth.java [hunk 20]
```diff
@@ -83,18 +84,17 @@ public Optional<RestApi> authenticate(Config config, Vault.Builder vaultBuilder)
...
-                        if (!Files.exists(tokenPath)) {
-                            return Optional.empty();
-                        }
-                        try {
-                            return Optional.of(Files.readString(tokenPath));
-                        } catch (IOException e) {
-                            throw new VaultApiException("Failed to read token from " + tokenPath.toAbsolutePath(), e);
-                        }
-                    });
+            Optional<String> maybeToken = config.get("auth.k8s.service-account-token").asString().or(() -> {
```
Comment (romain-grecourt) [inline integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sVaultAuth.java:87]: Seems like this change can be avoided.

File: integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java [hunk 21]
```diff
@@ -16,58 +16,149 @@
...
+            case PEM, PEM_BUNDLE -> mount + "/ca_chain";
+        };
+
+        VaultOptionalResponse<byte[]> response = restApi.getBytes(apiPath,
+                                                                  request,
+                                                                  VaultOptionalResponse.<byte[], byte[]>vaultResponseBuilder()
+                                                                          .entityProcessor(Function.identity()));
+        return CaCertificateGet.Response.builder()
+                .entity(response.entity().orElseThrow(() ->
+                                                              VaultRestException.builder()
```
Comment (romain-grecourt) [inline integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java:63]: Indentation seems excessive, can this be aligned with `response.entity` ?
I.e.

```java
.entity(response.entity().orElseThrow(() ->
        VaultRestException.builder()
                          .headers(response.headers())
                          .requestId(response.requestId())
                          .status(response.status())
                          .vaultErrors(response.errors())
                          .message(
                                  "CA Certificate is expected to be present, but is"
                                          + " not available")
                          .build()))
```

File: integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java [hunk 22]
```diff
@@ -16,58 +16,149 @@
...
+            case PEM, PEM_BUNDLE -> mount + "/crl/pem";
+        };
+
+        VaultOptionalResponse<byte[]> response = restApi.getBytes(apiPath,
+                                                                  request,
+                                                                  VaultOptionalResponse.<byte[], byte[]>vaultResponseBuilder()
+                                                                          .entityProcessor(Function.identity()));
+        return CrlGet.Response.builder()
+                .entity(response.entity().orElseThrow(() ->
+                                                              VaultRestException.builder()
```
Comment (romain-grecourt) [inline integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java:106]: Same comment as above (excessive indentation)

File: examples/integrations/vault/hcp/src/main/java/io/helidon/examples/integrations/vault/hcp/VaultMain.java [hunk 23]
```diff
@@ -58,8 +57,11 @@ public static void main(String[] args) {
         // 1. Token based authentication
         Vault tokenVault = Vault.builder()
                 .config(config.get("vault.token"))
-                .updateWebClient(it -> it.connectTimeout(5, TimeUnit.SECONDS)
-                        .readTimeout(5, TimeUnit.SECONDS))
+                .updateWebClient(client -> client.channelOptions(
```
[the remark this answers is not in the document]
Comment (arjav-desai) [reply examples/integrations/vault/hcp/src/main/java/io/helidon/examples/integrations/vault/hcp/VaultMain.java]: Updated

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java (hunk 18 above)
Comment (arjav-desai) [reply integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java]: Fixed.

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java (hunk 19 above)
Comment (arjav-desai) [reply integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sRestApi.java]: Fixed.

File: integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sVaultAuth.java (hunk 20 above)
Comment (arjav-desai) [reply integrations/vault/auths/k8s/src/main/java/io/helidon/integrations/vault/auths/k8s/K8sVaultAuth.java]: Fixed.

File: integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java (hunk 21 above)
Comment (arjav-desai) [reply integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java:63]: This is how the idea template is updating it so will touch base on this.

File: integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java (hunk 22 above)
Comment (arjav-desai) [reply integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java:106]: This is how the idea template is updating it so will touch base on this.

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java (hunk 17 above)
Comment (arjav-desai) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/RestApi.java]: Fixed.

File: integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiOptionalResponse.java [hunk 24]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Oracle and/or its affiliates.
+ * Copyright (c) 2021, 2023 Oracle and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (arjav-desai) [reply integrations/common/rest/src/main/java/io/helidon/integrations/common/rest/ApiOptionalResponse.java]: Fixed.

File: integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java (hunk 21 above)
Comment (arjav-desai) [reply integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java:63]: I tried updating it manually, but then doing reformat is putting it at same location again.

Comment (romain-grecourt) [reply integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java:63]: You can actually do it this way:

```java
return CaCertificateGet.Response.builder()
        .entity(response.entity()
                        .orElseThrow(() -> VaultRestException.builder()
                                .headers(response.headers())
                                .requestId(response.requestId())
                                .status(response.status())
                                .vaultErrors(response.errors())
                                .message(
                                        "CA Certificate is expected to be present, but is"
                                                + " not available")
                                .build()))
        .headers(response.headers())
        .requestId(response.requestId())
        .status(response.status())
        .build();
```

File: integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java (hunk 22 above)
Comment (romain-grecourt) [reply integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java:106]:
```java
return CrlGet.Response.builder()
        .entity(response.entity()
                        .orElseThrow(() -> VaultRestException.builder()
                                .headers(response.headers())
                                .requestId(response.requestId())
                                .status(response.status())
                                .vaultErrors(response.errors())
                                .message(
                                        "CRL is expected to be present, but is not "
                                                + "available")
                                .build()))
        .headers(response.headers())
        .requestId(response.requestId())
        .status(response.status())
        .build();
```

Comment (arjav-desai) [reply integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java]: updated it.

File: integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java (hunk 21 above)
Comment (arjav-desai) [reply integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java]: updated it.

**Review by romain-grecourt — APPROVED** [review]
